src/SFML/Window/Win32/JoystickImpl.cpp

```diff
@@ -44,1115 +44,724 @@
-    JOYINFOEX pos;
-    pos.dwFlags = JOY_RETURNX | JOY_RETURNY | JOY_RETURNZ | JOY_RETURNR | JOY_RETURNU | JOY_RETURNV | JOY_RETURNBUTTONS;
-    pos.dwFlags |= (m_caps.wCaps & JOYCAPS_POVCTS) ? JOY_RETURNPOVCTS : JOY_RETURNPOV;
-    pos.dwSize = sizeof(JOYINFOEX);
-    if (joyGetPosEx(m_index, &pos) == JOYERR_NOERROR)
+    if (joystickImpl.m_useXInput)
     {
-        // The joystick is connected
-        state.connected = true;
-
-        // Axes
-        state.axes[Joystick::Axis::X] = (static_cast<float>(pos.dwXpos) -
-                                         static_cast<float>(m_caps.wXmax + m_caps.wXmin) / 2.f) *
-                                        200.f / static_cast<float>(m_caps.wXmax - m_caps.wXmin);
-        state.axes[Joystick::Axis::Y] = (static_cast<float>(pos.dwYpos) -
-                                         static_cast<float>(m_caps.wYmax + m_caps.wYmin) / 2.f) *
-                                        200.f / static_cast<float>(m_caps.wYmax - m_caps.wYmin);
-        state.axes[Joystick::Axis::Z] = (static_cast<float>(pos.dwZpos) -
-                                         static_cast<float>(m_caps.wZmax + m_caps.wZmin) / 2.f) *
-                                        200.f / static_cast<float>(m_caps.wZmax - m_caps.wZmin);
-        state.axes[Joystick::Axis::R] = (static_cast<float>(pos.dwRpos) -
-                                         static_cast<float>(m_caps.wRmax + m_caps.wRmin) / 2.f) *
-                                        200.f / static_cast<float>(m_caps.wRmax - m_caps.wRmin);
-        state.axes[Joystick::Axis::U] = (static_cast<float>(pos.dwUpos) -
-                                         static_cast<float>(m_caps.wUmax + m_caps.wUmin) / 2.f) *
-                                        200.f / static_cast<float>(m_caps.wUmax - m_caps.wUmin);
-        state.axes[Joystick::Axis::V] = (static_cast<float>(pos.dwVpos) -
-                                         static_cast<float>(m_caps.wVmax + m_caps.wVmin) / 2.f) *
-                                        200.f / static_cast<float>(m_caps.wVmax - m_caps.wVmin);
-
-        // Special case for POV, it is given as an angle
-        if (pos.dwPOV != 0xFFFF)
+        // XInput has 14 Buttons and 6 Axes
+        joystickImpl.m_caps.buttonCount = 14;
+        constexpr auto axes             = 6;
+        for (unsigned int i = 0; i < axes; ++i)
+            joystickImpl.m_caps.axes[getAxis(static_cast<int>(i))] = true;
+    }
+    else
+    {
+        // RawInput query capabilities
+        UINT buffSize{};
+        if (const auto result = GetRawInputDeviceInfoW(deviceHandle, RIDI_PREPARSEDDATA, nullptr, &buffSize);
+            FAILED(result))
         {
-            const float angle                = static_cast<float>(pos.dwPOV) / 18000.f * 3.141592654f;
-            state.axes[Joystick::Axis::PovX] = std::sin(angle) * 100;
-            state.axes[Joystick::Axis::PovY] = std::cos(angle) * 100;
+            // that didn't work!
```

Comment:
```suggestion
```
I appreciate the comment but I think the `FAILED(result)` right above this makes it clear enough that this condition did not succeed.